### PR TITLE
⏳ Early Enhancement & Pre-Audit Release

### PR DESCRIPTION
**Pull request by JaredBorders**
Added two validated function selectors to the SMv3 module and drastically improved testing.
Deployed pre-audit, optimized validation modules to OP, OP Goerli, Base, and Base Goerli.

## Description
* Add new valid selectors to `SMv3SessionValidationModule`: `depositEth`, `withdrawEth`
* Refactor/optimize logic in both modules
* Refactor tests (drastic improvement)

## Related issue(s)
N/A

## Motivation and Context
Deploy early version for internal testing on mainnet 😮‍💨 (and add/improve things already discovered internally that will surely be addressed by the auditors)

### CODE REVIEW

**Comment by JaredBorders** [comment]
@insulineru tagging you just for visibility. you may be interested in the optimizations, etc

File: src/SMv3SessionValidationModule.sol [hunk 1]
```diff
@@ -38,36 +34,34 @@ contract SMv3SessionValidationModule is ISessionValidationModule {
...
         }
 
-        /// @dev ensure call value is zero
-        if (callValue != 0) {
+        /// @dev ensure call value is zero unless calling IEngine.depositEth
+        if (funcSelector == IEngine.depositEth.selector) {
+            if (callValue == 0) {
+                revert InvalidCallValue();
+            }
+        } else if (callValue != 0) {
```
Comment (jcmonte) [inline src/SMv3SessionValidationModule.sol:64]: Can we do something like this for SMv2?

Comment (JaredBorders) [reply src/SMv3SessionValidationModule.sol:64]: Yes!

see https://github.com/Kwenta/scw-contracts/pull/1/commits/740f28340285500382d262ea397660bf19682bf0

File: src/SMv2SessionValidationModule.sol [hunk 2]
```diff
@@ -23,45 +23,29 @@ contract SMv2SessionValidationModule is ISessionValidationModule {
...
      * complies with the Session Key permissions represented by sessionKeyData
      * @param destinationContract address of the contract to be called
-     * @param callValue value to be sent with the call
      * @param _funcCallData the data for the call. is parsed inside the SVM
      * @param _sessionKeyData SessionKey data, that describes sessionKey permissions
      */
     function validateSessionParams(
         address destinationContract,
-        uint256 callValue,
+        uint256, /*callValue*/
```
Comment (jcmonte) [inline src/SMv2SessionValidationModule.sol:31]: TIL

**Review by jcmonte — APPROVED** [review]

**Pull request merged**